compiler-hacking.md

```diff
@@ -0,0 +1,464 @@
+# Understanding the Shemsu compiler
+This is a guide to understanding the Shesmu olive compiler. This is meant to
+explain the core components that get the language to Java bytecode.
```

Comment:
Could you detail what background knowledge (JVM, etc) is needed to understand the document?